test/src/lib.rs

```diff
@@ -45,7 +43,7 @@ mod tests {
         let conn = tmp_db.connect_limbo();
 
         let list_query = "SELECT * FROM test";
-        let max_iterations = 10000;
+        let max_iterations = 1000;
```

Comment:
Please don't change this, changing iterations modifies what the functions is testing (btree page splitting). Is there a reason for this change? If not let's revert

Reply:
@pereman2 if you were to add a code comment explaining that in the test, what would it be

Reply:
Is using the same test db for each test also part of the btree page splitting test logic btw?

Reply:
>Please don't change this, changing iterations modifies what the functions is testing (btree page splitting). Is there a reason for this change? If not let's revert

It basically took 2 minutes on my 10 core 64 GB mac(though my IDE setting uses less). 
<img width="1628" alt="image" src="https://github.com/user-attachments/assets/14cfb7b2-62bb-4690-95b8-5197b1f73f1a">

Is there a way to test btree page splitting with less iterations?

Reply:
> @pereman2 if you were to add a code comment explaining that in the test, what would it be

will do it next day probably, can't today.

> Is using the same test db for each test also part of the btree page splitting test logic btw?

Right know I regenerate the test.db file for every test so that it starts fresh.

> It basically took 2 minutes on my 10 core 64 GB mac(though my IDE setting uses less).
yes because I'm doing a rigorous per iteration check of all values which is quite slow, that's true.


> Is there a way to test btree page splitting with less iterations?

yes of course, I just put a big number that I knew it would trigger to page splits: one for root page and one for a a child page. I can take the time to calculate it in the future.

> ➕ Should we verify b-tree table splitting in this test? At first glance, I thought the purpose of this test is to verify whether writes are working well(as the name test_sequential_write suggests). Maybe we can add other tests which verify b-tree table splitting.

We could for sure, but right now these tests are made like these for velocity of development right now. I'm still early in the development of btree logic and I'm not sure of everything.

Reply:
Got the point. Then let me just preserve the `max_iteration` for now.

Reply:
I've added a [PR](https://github.com/penberg/limbo/pull/330) which adds a test to check page splitting(root and leaf). PTAL 🙏